pkg/kubectl/cmd/util/helpers.go

```diff
@@ -621,3 +626,55 @@ func ShouldIncludeUninitialized(cmd *cobra.Command, includeUninitialized bool) b
 	}
 	return shouldIncludeUninitialized
 }
+
+// DescriberFunc gives a way to display the specified RESTMapping type
+type DescriberFunc func(restClientGetter genericclioptions.RESTClientGetter, mapping *meta.RESTMapping) (printers.Describer, error)
+
+// DescriberFn gives a way to easily override the function for unit testing if needed
+var DescriberFn DescriberFunc = describer
+
+// Returns a Describer for displaying the specified RESTMapping type or an error.
+func describer(restClientGetter genericclioptions.RESTClientGetter, mapping *meta.RESTMapping) (printers.Describer, error) {
+	clientConfig, err := restClientGetter.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+	// try to get a describer
+	if describer, ok := printersinternal.DescriberFor(mapping.GroupVersionKind.GroupKind(), clientConfig); ok {
+		return describer.(printers.Describer), nil
+	}
+	// if this is a kind we don't have a describer for yet, go generic if possible
+	if genericDescriber, genericErr := genericDescriber(restClientGetter, mapping); genericErr == nil {
+		return genericDescriber, nil
+	}
+	// otherwise return an unregistered error
+	return nil, fmt.Errorf("no description has been implemented for %s", mapping.GroupVersionKind.String())
+}
+
+// helper function to make a generic describer, or return an error
+func genericDescriber(restClientGetter genericclioptions.RESTClientGetter, mapping *meta.RESTMapping) (printers.Describer, error) {
+	clientConfig, err := restClientGetter.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	// used to fetch the resource
+	dynamicClient, err := dynamic.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := restClientGetter.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	// used to get events for the resource
+	clientSet, err := internalclientset.NewForConfig(config)
```

Comment:
@deads2k need to use the fake RESTClient here, but I don't believe adding a new `ToClientSet` method to ConfigFlags is the correct solution

Reply:
> @deads2k need to use the fake RESTClient here, but I don't believe adding a new ToClientSet method to ConfigFlags is the correct solution

You are correct.  ClientSets are not generic.  Why does it need to be fake?  This look acceptable for kubectl for the time being.